source/system-administrators/activities/language-support/add-new-language.rst

```diff
@@ -0,0 +1,152 @@
+
+    .. code-block:: js
+       :caption: *studio-ui/static-assets/components/cstudio-common/resources/ja/base.js*
+       :linenos:
+
+       CStudioAuthoring.Messages.registerBundle("siteDashboard", "ja", {
+       dashboardTitle: "ダッシュボード",
+
+       dashboardCollapseAll: "Collapse All",
+       ...
+
+Remember to change the language code in the all the ``registerBundle`` calls in the ``base.js`` file
+
+  .. code-block:: js
+
+     CStudioAuthoring.Messages.registerBundle("dialogs", "ja", {
+
+----------------------------------------------------------
+3. Add the new language to the get-available-languages API
+----------------------------------------------------------
+
+* To add the new language to the ``get-available-languages`` API (*/studio/api/1/services/api/1/server/get-available-languages.json*), in your ``studio`` code, navigate to ``studio/src/main/webapp/default-site/scripts/rest/api/1/server`` and open the ``get-available-languages.get.groovy`` file
+* Add the new language to the file:
+
+  .. code-block:: groovy
+     :emphasize-lines: 14-16
+     :linenos:
+
+     def result = []
+    	result[0] = [:]
+    	result[0].id = "en"
+    	result[0].label = "English"
+    	result[1] = [:]
+    	result[1].id = "es"
+    	result[1].label = "español"
+    	result[2] = [:]
+    	result[2].id = "kr"
+    	result[2].label = "한국어"
+    	result[3] = [:]
+    	result[3].id = "de"
+    	result[3].label = "Deutsch"
+        result[4] = [:]
+        result[4].id = "ja"
+        result[4].label = "日本語"
+     return result
+
+---------------------------------------------------------
+4. Add the new language to the react translations manager
+---------------------------------------------------------
+* To add the new language to the react translations manager, in your ``studio-ui`` code, navigate to ``studio-ui/ui/app/scripts`` folder then open the file ``i18n.js``
+* Add the new language code to the ``languages`` list and save your changes
+
+  .. code-block:: js
+     :emphasize-lines: 5
+     :linenos:
+     :caption: *studio-ui/ui/app/scripts/i18n.js*
+
+     manageTranslations({
+       messagesDirectory: './src/translations/src',
+       translationsDirectory: './src/translations/locales/',
+       whitelistsDirectory: './src/translations/whitelists',
+       languages: ['en', 'es', 'de', 'ko', 'ja']
+     });
+
+* Open the command line and navigate to ``studio-ui/ui/app/scripts`` folder and run the following commands to update and generate the ``ja.json`` file:
+
+  * *yarn i18n:extract*
+  * *yarn i18n:manage*
+
+  |
+
+  Here's some of the output when running the above commands:
+
+  .. code-block:: bash
+     :linenos:
+     :emphasize-lines: 1,5
+
+     ➜  scripts git:(develop) ✗ yarn i18n:extract
+     yarn run v1.13.0
+     $ NODE_ENV=production babel ./src --extensions '.ts,.tsx' --out-file /dev/null
+     ✨  Done in 3.15s.
+     ➜  scripts git:(develop) ✗ yarn i18n:manage
+     yarn run v1.13.0
+     $ node scripts/i18n.js
```

Comment:
Once the scripts run, people need to open the newly generated locale file (i.e. src/translations/locales/ja.json) and translate it. The file will initially be populated with the default messages (English).